### PR TITLE
feat: add ability to choose CNI config

### PR DESCRIPTION
**Pull request by Unix4ever**
Initial version which only allows setting CNI using preset, no custom
CNI urls are supported at the moment. Still need to figure out what kind
of UI can be used for that.

Signed-off-by: Artem Chernyshev <artem.0xD2@gmail.com>

### CODE REVIEW

**Comment by Unix4ever** [comment]
How it looks like:
image

File: api/machine/machine.proto [hunk 1]
```diff
@@ -747,8 +747,14 @@ message ControlPlaneConfig {
   string endpoint = 1;
 }
 
+message CNIConfig {
+  string name = 1;
+  repeated string urls = 2;
+}
```
Comment (andrewrynhard) [inline api/machine/machine.proto:753]: I wonder if one day we can generate protobufs from our go structs so that the two are always in sync.

**Review by andrewrynhard — APPROVED** [review]
This should probably be shown only for `init` nodes.

File: api/machine/machine.proto (hunk 1 above)
Comment (Unix4ever) [reply api/machine/machine.proto:753]: yeah, that can be nice... Do we have any existing means to generate protobufs from go structs?

**Review by smira — APPROVED** [review]

**Pull request merged**